src/compiler.rs

```diff
@@ -48,7 +51,12 @@ impl Compiler {
             let parent = current.path.parent().unwrap();
 
             let import = if let Some(relative) = relative {
-              let path = parent.join(Self::expand_tilde(&relative.cooked)?);
+              let expanded_path = match full(&relative.cooked) {
+                Ok(value) => value,
+                Err(error) => return Err(Error::UnresolvedVariableInModule { variable: error.var_name, module: *name }),
+              };
```

Comment:
You can do something like this:
```suggestion
              let expanded_path = full(&relative.cooked).map_err(|error| Error::UnresolvedVariableInModule { variable: error.var_name, module: *name }))?
```